File: sdks/python-mcp/tests/mcp_authoring/repo_paths.py
from __future__ import annotations

from pathlib import Path

import yaml


def lookup_mcp_fixtures() -> Path:
    here = Path(__file__).resolve()
    for parent in here.parents:
        manifest = parent / "contract" / "manifest" / "repo-paths.yaml"
        if manifest.is_file():
            loaded = yaml.safe_load(manifest.read_text())
            if not isinstance(loaded, dict):
                raise RuntimeError("repo-paths.yaml is not a mapping")
            lookups = loaded.get("lookups")
            if not isinstance(lookups, dict) or "mcpFixtures" not in lookups:
                raise RuntimeError("lookups.mcpFixtures is missing from repo-paths.yaml")
            rel = lookups["mcpFixtures"]
            if not isinstance(rel, str) or not rel:
                raise RuntimeError("lookups.mcpFixtures must be a non-empty string")
            return parent / rel
    raise RuntimeError("could not locate contract/manifest/repo-paths.yaml")

File: sdks/ruby-mcp/test/mcp_authoring/repo_paths.rb
# frozen_string_literal: true

require "pathname"
require "yaml"

module McpAuthoring
  module RepoPaths
    module_function

    def lookup_mcp_fixtures
      here = Pathname.new(__dir__).expand_path
      here.ascend do |parent|
        manifest = parent.join("contract", "manifest", "repo-paths.yaml")
        next unless manifest.file?

        loaded = YAML.safe_load(manifest.read)
        raise "repo-paths.yaml is not a mapping" unless loaded.is_a?(Hash)

        lookups = loaded["lookups"]
        unless lookups.is_a?(Hash) && lookups.key?("mcpFixtures")
          raise "lookups.mcpFixtures is missing from repo-paths.yaml"
        end

        rel = lookups["mcpFixtures"]
        raise "lookups.mcpFixtures must be a non-empty string" unless rel.is_a?(String) && !rel.empty?

        return parent.join(rel)
      end
      raise "could not locate contract/manifest/repo-paths.yaml"
    end
  end
end
